fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+
+    fetch('/api/todos')
+      .then(response => response.json())
+      .then(responseData => this.setState({ todos: responseData }))
+      .catch(error => console.error('Fetch error', error));
+  }
+
+  addTodo(newTodo) {
+
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(response => response.json())
+      .then(responseData => {
+        this.state.todos.push(responseData);
+        this.setState([{ todos: responseData }]);
+      })
```

Comment:
You'll want to make a copy of the original array, add the response todo, then call setState to replace the old array

```js
this.setState({ todos: newArray })
```

check out `Array.prototype.concat` for this.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat